src/cmem/ActivityControl/SilkActivityControl.tsx

```diff
@@ -167,38 +186,125 @@ export function useSilkActivityControl({
     layoutConfig = defaultLayout,
     hideMessageOnStatus = () => false,
     executePrioritized,
+    translateUnits, 
     ...props
 }: SilkActivityControlProps) {
     const [activityStatus, setActivityStatus] = useState<SilkActivityStatusProps | undefined>(initialStatus);
-    const currentStatus = useRef<SilkActivityStatusProps | undefined>(initialStatus)
-    const [showStartPrioritized, setShowStartPrioritized] = useState(false)
+    const currentStatus = useRef<SilkActivityStatusProps | undefined>(initialStatus);
+    const [showStartPrioritized, setShowStartPrioritized] = useState(false);
     const [errorReport, setErrorReport] = useState<string | SilkActivityExecutionReportProps | undefined>(undefined);
+    const [statusTimers, setStatusTimers] = React.useState<StatusTimers>({ ...defaultStatusTimers });
+    const runningTimerRef = React.useRef<NodeJS.Timer>();
```

Comment:
It's probably not necessary to distinguish running and waiting time. Also see comments below.